Integrate shared table collapsing js into iOS app

montehurd opened the pull request:
https://phabricator.wikimedia.org/T158686

The first commit allows us to easily pull in common js (and other) bits used by both iOS and Android.
Stephen set up repro for these common bits here: https://github.com/wikimedia/applib

The second commit fixes an error which was happening when we ran Grunt (which processes all of the files in the "www" directory and generates the "assets" directory files which get bundled).

niedzielski commented on www/package.json:
```diff
@@ -3,6 +3,9 @@
   "repository": "https://gerrit.wikimedia.org/r/apps/ios/wikipedia",
   "version": "0.1.0",
+  "dependencies": {
+       "applib": "0.1.2"
```
Consider merging [this guy](https://github.com/wikimedia/applib/pull/1) and then you can use the new API

montehurd replied:
Done!

niedzielski commented:
Since adding an NPM depenedency will have no functional change in itself, please consider adding the file changes for the to `collapseTables.js` and the generated Grunt bundle as well

montehurd commented:
@niedzielski I was hoping to keep follow-on functional changes separate to simplify shepherding this PR through code review.

Oh was [this](https://phabricator.wikimedia.org/T158309#3059605) the ticket pattern you suggested, or does it sound good?

niedzielski commented:
:+1: @montehurd whatever makes sense as long as it's coming!

montehurd commented:
@niedzielski 

Ok I updated the title of https://phabricator.wikimedia.org/T158686 to...

"Integrate shared **table collapsing** js into iOS app" 

... and I added a commit to this PR integrating the 1st method "getTableHeader".

montehurd commented on www/js/transforms/collapseTables.js:
```diff
@@ -1,42 +1,6 @@
-    }
-    return thArray;
-}
+var getTableHeader = require("applib").CollapseElement.getTableHeader;
```
`getTableHeader` was the first method we moved to the common [applib](https://github.com/wikimedia/applib) repo 

We then removed `getTableHeader` from our `www/js/transforms/collapseTables.js` iOS source file, then ran `make grunt` to rebuild the bundled `assets` folder contents and confirmed the version we removed disappeared from `Wikipedia/assets/index.js` and the version we added to [applib](https://github.com/wikimedia/applib) appeared in `Wikipedia/assets/index.js`.

montehurd commented on www/package.json:
```diff
@@ -4,7 +4,7 @@
   "version": "0.1.0",
   "dependencies": {
-       "applib": "0.1.2"
+       "applib": "1.0.0"
```
I updated `applib` version to `1.0.0`, then ran `make grunt`, which caused `Wikipedia/assets/index.js` to be rebuilt with upstream changes.

montehurd commented on www/js/transforms/collapseTables.js:
```diff
@@ -1,6 +1,6 @@
 var transformer = require("../transformer");
 var utilities = require("../utilities");
-var getTableHeader = require("applib").CollapseElement.getTableHeader;
+var getTableHeader = require("applib").CollapseTable.getTableHeader;
```
Here I updated our source to reference the name change from the last commit. I then ran `make grunt` again to regenerate `Wikipedia/assets/index.js`.

montehurd commented:
Closing this because I need to update some our "devDependencies" versions. Will do the bits from this PR as follow-ons.
